src/python/pants/backend/python/target_types_rules.py

```diff
@@ -272,12 +273,47 @@ async def infer_pex_binary_entry_point_dependency(
     maybe_disambiguated = explicitly_provided_deps.disambiguated(
         owners.ambiguous, owners_must_be_ancestors=entry_point.file_name_used
     )
-    unambiguous_owners = owners.unambiguous or (
-        (maybe_disambiguated,) if maybe_disambiguated else ()
-    )
+
+    if owners.unambiguous:
+        unambiguous_owners = owners.unambiguous
+    elif maybe_disambiguated:
+        unambiguous_owners = (maybe_disambiguated,)
+    elif owners.ambiguous and not maybe_disambiguated:
+        _handle_unresolved_pex_entrypoint(
+            address,
+            entry_point_field.value,
+            owners.ambiguous,
+            python_infer_subsystem.unowned_dependency_behavior,
+        )
+        unambiguous_owners = ()
+    else:
+        unambiguous_owners = ()
```

Comment:
Consider it personal preference, but when chaining `if`/`elif` for setting a variable, I prefer a helper function.
- `return` the value
- You can type hint it
- Help ensure the value is always set
- Makes the control flow a little more obvious

Reply:
That's a good way to think about it! I'll be adding this to my personal style as well.